### PR TITLE
Stock model

### PR DESCRIPTION
**Pull request by rayrobdod**
Basically, Issue #7.

### CODE REVIEW

**Comment by mikairi** [comment]
What you are doing is more like Java. The C# way to do properties, for example with `Stock.CurrentPrice`:

``` csharp
public decimal CurrentPrice { get; private set; }
```

If you have a property with a default value like `Note`:

``` csharp
public string Note { get; set; } = "";
```

You don't need those private fields at all. Your code looks cleaner and more concise this way.

Also it's just my personal preference, but `PriceAtTransactionTime` is a little too wordy, while `Price` would practically mean the same thing in the context of a `StockTransaction`. But other than that :+1:

**Comment by rayrobdod** [comment]
[Auto-Implemented Properties](https://msdn.microsoft.com/en-us/library/bb384054.aspx). That wasn't the easiest thing to find. Today I learned.

**Comment by mikairi** [comment]
Since no one else has any problem, I'm merging this.
